example/advection/advection.hpp

```diff
@@ -0,0 +1,94 @@
+//========================================================================================
+// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef EXAMPLE_ADVECTION_ADVECTION_HPP_
+#define EXAMPLE_ADVECTION_ADVECTION_HPP_
+
+#include <memory>
+
+#include "driver/driver.hpp"
+#include "driver/multistage.hpp"
+#include "interface/Container.hpp"
+#include "interface/StateDescriptor.hpp"
+#include "mesh/mesh.hpp"
+#include "task_list/tasks.hpp"
+
+using parthenon::MultiStageBlockTaskDriver;
+using parthenon::MeshBlock;
+using parthenon::Mesh;
+using parthenon::Outputs;
+using parthenon::BaseTask;
+using parthenon::Real;
+using parthenon::CellVariable;
+using parthenon::StateDescriptor;
+using parthenon::TaskStatus;
+using parthenon::TaskList;
+using parthenon::TaskListStatus;
+using parthenon::TaskID;
+using parthenon::Metadata;
+using parthenon::Params;
+using parthenon::Container;
+using parthenon::ParameterInput;
+using parthenon::ParArrayND;
+using parthenon::Integrator;
+using parthenon::BlockStageNamesIntegratorTaskFunc;
+using parthenon::BlockStageNamesIntegratorTask;
+using parthenon::BlockTaskFunc;
+using parthenon::BlockTask;
+using parthenon::BoundaryValues;
+using parthenon::AmrTag;
```

Comment:
At some point we should consider providing an include file that has a standard set of includes in something like packageInclude.hpp so that this code doesn't have to be replicated for every package.

Reply:
I'm not sure I agree. I don't like generic includes, as they hide information. I think this boiler plate has some value as a comment.

Reply:
The way Rust solves this is they have a "prelude" module for common names. So we could perhaps have something like:

```c++
namespace parthenon { namespace prelude {
// Re-exports
using Real = ::parthenon::Real;
using MeshBlock = ::parthenon::MeshBlock;
using ParArrayND = ::parthenon::PayArrayND;
} }
```

And then in the user code, you could write:
```c++
using namespace parthenon::prelude;
```

You should be careful, though, to keep that `prelude` namespace fairly lean - only things that are used by essentially ever single user. You could potentially split the preludes into something like `parthenon::task::prelude` and `parthenon::app::prelude` to distinguish being names commonly used by apps and names commonly used by task implementations.

Reply:
> I'm not sure I agree. I don't like generic includes, as they hide information. I think this boiler plate has some value as a comment.

Then why not just use everything from namespace parthenon? When the list of entities used from the namespace exceeds five or six, it starts to get unwieldy.  This would be no different than including a header file. My fear is that at some point folks will just copy and paste the entire 'using' list as their starting point and not prune it down to the ones used.  

I like the idea of splitting parthenon into sub-spaces and using those.  Some subsets that seem useful are Mesh, AMR, Task, and Types.

Reply:
Made an issue: #100.